### DISCUSSION TITLE
Flunt retornando Notifications mesmo quando cumpre os requisitos

### DISCUSSION THREAD

**Discussion by Gml10** [balta-io/forum#377] [Tira Dúvidas] [answered]
Tenho o seguinte código em uma entidade:

`public DeviceSystem(string model, string Identifier, string? name = null, string? description = null)
        {
            name ??= "New Device";

            var contract = new Flunt.Validations.Contract<Notification>()
                //model
                .IsNullOrWhiteSpace(model, "modelIsEmpty", "Informe o modelo do dispositivo")
                .IsGreaterThan(model, 50, "modelIsGreater", "O modelo do dispositivo deve conter no máximo 50 caracteres")
                //identifier
                .IsNullOrWhiteSpace(Identifier, "identifierIsEmpty", "Informe o identificador único do dispositivo")
                .IsGreaterThan(Identifier, 50, "identifierIsGreater", "O identificador único do dispositivo deve conter no máximo 64 caracteres")
                //name
                .IsGreaterThan(name, 64, "nameIsGreater", "O identificador único do dispositivo deve conter no máximo 64 caracteres")
                //description
                .IsGreaterThan(name, 250, "descriptionIsGreater", "A descrição do dispositivo deve conter no máximo 250 caracteres");

            AddNotifications(
                contract
            );
            this.Model = model;
            this.Identifier = Identifier;
            this.Name = name;
            this.Description = description;
        }`
        
No debug verifiquei que todas as propriedades estão sendo recebidas no construtor corretamente e dentro das validações, porem todas as 6 notificações são adicionadas na lista de notifications o que resulta em IsValid = false. Estou fazendo algo de errado? Outra coisa qual o propósito de passar um tipo ao instanciar o Contract?

EDIT: Agora entendi o meu erro. Quando informo IsNullOrWhiteSpace estou dizendo que a propriedade DEVE SER vazia ou em branco não que ser nula e/ou vazia é a condição para dispara um erro. Isso é um pouco confuso hehehe

**Comment by andrebaltieri** [comment, marked as answer]
Show, que bom que resolveu @Gml10 !!
